### ISSUE TITLE
Add .jsx and .tsx support

### ISSUE THREAD

**Issue by phocks** [abcnews/aunty#200]
It seems to be becoming a standard now that any JavaScript and TypeScript files that contain JSX syntax have the full `.jsx` or `.tsx` so I'm suggesting we update our React and Preact templates to use these filenames and add import support to WebPack (I don't think it handles these file types currently).

**Comment by drzax** [comment]
I thought they were already? https://github.com/abcnews/aunty/blob/main/src/generators/project/templates/react/src/components/App/index.tsx

**Comment by phocks** [comment]
Ahh yes my mistake .tsx seems to be implemented, but not .jsx

Probably not overly urgent, but .jsx support could be sort of easy to implement

I also noticed a strange bug where if you generate a project and NOT choose TypeScript it adds this in your index.js file which includes `: any` which is typescript. Small bug, I'll track it down and open another issue.

```javascript
if (module.hot) {
  module.hot.accept('./components/App', () => {
    try {
      renderApp();
    } catch (err: any) {
      import('./components/ErrorBox').then(({ default: ErrorBox }) => {
        root.render(<ErrorBox error={err} />);
      });
    }
  });
}
```

**Comment by drzax** [comment]
Yeah, so you'd just rename with the new extension if `isTS` is false? Sounds like a good idea to me.

**Comment by Fe1777** [comment]
Let's outline a detailed framework to add .jsx support to the REWS abcnews/aunty project while considering that .tsx support is already implemented. This plan will be divided into key steps for clarity.

### Framework to Add .jsx Support

#### 1. **Pre-requisites and Initial Setup**
   - Ensure you have the project cloned and all dependencies installed.
   - Familiarize yourself with the existing project structure, especially focusing on how .tsx files are handled.

#### 2. **Update Webpack Configuration**
   - Modify the webpack config to recognize .jsx files:
     ```javascript
     module.exports = {
         // existing config...
         resolve: {
             extensions: ['.js', '.jsx', '.ts', '.tsx'], // Add .jsx here
         },
         module: {
             rules: [
                 // existing rules...
                 {
                     test: /\.(js|jsx)$/,
                     exclude: /node_modules/,
                     use: {
                         loader: 'babel-loader',
                         options: {
                             presets: ['@babel/preset-react', '@babel/preset-env']
                         }
                     }
                 },
                 // Handle .tsx files if not already added
                 {
                     test: /\.(ts|tsx)$/,
                     exclude: /node_modules/,
                     use: {
                         loader: 'ts-loader',
                         options: {
                             transpileOnly: true
                         }
                     }
                 }
             ],
         },
     };
     ```

#### 3. **Configure Babel**
   - Ensure Babel supports JSX:
     Create or update the `.babelrc` file:
     ```json
     {
         "presets": [
             "@babel/preset-env",
             "@babel/preset-react",
             "@babel/preset-typescript"
         ]
     }
     ```

#### 4. **Verify TypeScript Configuration**
   - Update `tsconfig.json` to include `.jsx` in `include` or `files` sections:
     ```json
     {
         "compilerOptions": {
             // ...existing options
             "jsx": "react"
         },
         "include": [
             "src/**/*.ts",
             "src/**/*.tsx",
             "src/**/*.js",  // Add .js
             "src/**/*.jsx"  // Add .jsx
         ]
     }
     ```

#### 5. **Code Updates and Linting**
   - Ensure that ESLint (if used) supports .jsx files:
     Update `.eslintrc`:
     ```json
     {
         "parser": "@typescript-eslint/parser",
         "extends": [
             "plugin:react/recommended",
             "plugin:@typescript-eslint/recommended"
         ],
         "settings": {
             "react": {
                 "version": "detect"
             }
         },
         "rules": {
             // existing rules
         }
     }
     ```

#### 6. **Testing and Validation**
   - Write sample .jsx files for testing purposes.
   - Run the development server and build scripts to ensure that the build system correctly processes .jsx files.
   - Perform end-to-end testing to confirm that there are no unexpected issues.

#### 7. **Documentation Updates**
   - Update project documentation to reflect the new .jsx support.
   - Mention any new dependencies, changes to configuration files, or special instructions.
   - Notify the team about the changes via the project's communication channels.

### Conclusion
By following this framework, you will ensure .jsx support is seamlessly integrated into the REWS abcnews/aunty project. It covers updating configurations, validating settings, testing the integration, and updating related documentation. 

#### Next Steps
1. Implement the changes as detailed.
2. Test thoroughly.
3. Document and notify the team.

This structured approach ensures that the new functionality is integrated without disrupting existing operations, particularly with the already supported .tsx files.
